components/org.wso2.carbon.identity.api.server.webhook.metadata/org.wso2.carbon.identity.api.server.webhook.metadata.v1/src/main/resources/webhook-metadata.yaml

```diff
@@ -0,0 +1,158 @@
+      operationId: getEventSchemas
+      tags:
+        - Webhooks
+      responses:
+        '200':
+          description: OK
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/WebhookMetadata'
+        '401':
+          description: Unauthorized
+        '403':
+          description: Forbidden
+        '500':
+          description: Internal server error
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Error'
+        x-codeSamples:
+          - lang: Curl
+            source: |
+              curl --location 'https://localhost:9443/t/carbon.super/api/server/v1/webhooks/event-schemas' -H 'Authorization: Basic YWRtaW46YWRtaW4='
+
+  /webhooks/metadata/event-schemas/{schemaName}/events:
+    get:
+      summary: List Events of Schema
+      description: "This API returns the list of events under the given event schema.\n\n <b>Scope(Permission) required:</b> `internal_webhook_metadata_view` \n\n"
+      operationId: getEventsBySchema
+      tags:
+        - Webhooks
+      parameters:
+        - name: schemaName
+          in: path
+          required: true
+          schema:
+            type: string
+      responses:
+        '200':
+          description: OK
+          content:
+            application/json:
+              schema:
+                type: array
+                items:
+                  $ref: '#/components/schemas/EventMetadata'
+        '400':
+          description: Bad Request
+        '401':
+          description: Unauthorized
+        '403':
+          description: Forbidden
+        '404':
+          description: Schema not found
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Error'
+        '500':
+          description: Internal server error
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Error'
+        x-codeSamples:
+          - lang: Curl
+            source: |
+              curl --location 'https://localhost:9443/t/carbon.super/api/server/v1/webhooks/event-schemas/WSO2/events' -H 'Authorization: Basic YWRtaW46YWRtaW4='
+
+components:
+  securitySchemes:
+    BasicAuth:
+      type: http
+      scheme: basic
+    OAuth2:
+      type: oauth2
+      flows:
+        authorizationCode:
+          authorizationUrl: 'https://localhost:9443/oauth2/authorize'
+          tokenUrl: 'https://localhost:9443/oauth2/token'
+          scopes:
+            internal_webhook_view: View webhook metadata
+
+  schemas:
+    EventSchema:
+      type: object
+      properties:
+        name:
+          type: string
+          example: WSO2
+        uri:
+          type: string
+          example: https://schemas.identity.wso2.org/events
```

Comment:
we should also include a property like 'ref' locating the uri from which the event schema can be retrieved further.

Reply:
eventSchemas -> eventProfile